python/data_extraction/export_data.py

```diff
@@ -1,63 +1,78 @@
 from data_extraction import content_export
 from data_extraction import taxonomy_query, plek
-import json
-import sys
+from data_extraction import json_arrays
+from data_extraction.helpers import dig
 import functools
 from multiprocessing import Pool
+import gzip
+import json
+import os
+
 
+config_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), '..', 'config', 'data_export_fields.json')
 
-with open('config/data_export_fields.json') as json_data_file:
+with open(config_path) as json_data_file:
     configuration = json.load(json_data_file)
 
 
-content_links_generator = content_export.content_links_generator(
-    blacklist_document_types=configuration['blacklist_document_types']
-)
+def __transform_content(input_filename="data/content.json.gz",
+                        output_filename="data/filtered_content.json.gz",
+                        transform_function=lambda x: x):
+    with gzip.open(input_filename, mode='rt') as input_file:
+        with gzip.open(output_filename, mode='wt') as output_file:
+            content_generator = json_arrays.read_json(input_file)
+            json_arrays.write_json(output_file, transform_function(content_generator))
 
 
-slicer = functools.partial(content_export.content_hash_slicer,
-                           base_fields=configuration['base_fields'],
-                           taxon_fields=configuration['taxon_fields'],
-                           ppo_fields=configuration['ppo_fields'])
+def export_content(output_filename="data/content.json"):
+    def __complete_content():
+        get_content = functools.partial(content_export.get_content,
+                                        content_store_url=plek.find('draft-content-store'))
 
+        content_links_generator = content_export.content_links_generator(
+            blacklist_document_types=configuration['blacklist_document_types']
+        )
 
-get_content = functools.partial(content_export.get_content,
-                                content_store_url=plek.find('draft-content-store'))
+        pool = Pool(10)
+        return pool.imap(get_content, content_links_generator)
 
+    with gzip.open(output_filename, 'wt') as output_file:
+        json_arrays.write_json(output_file,
+                               filter(lambda link: link, __complete_content()))
 
-def taxonomy():
-    query = taxonomy_query.TaxonomyQuery()
-    level_one_taxons = query.level_one_taxons()
-    children = [taxon
-                for level_one_taxon in level_one_taxons
-                for taxon in query.child_taxons(level_one_taxon['base_path'])]
-    return iter(level_one_taxons + children)
 
+def export_filtered_content(input_filename="data/content.json.gz", output_filename="data/filtered_content.json.gz"):
```

Comment:
It would be useful if the name for this file made it clear what it contains. What is the filtering in this case?

Reply:
tricky to say because it may change. It filters a bunch of attributes of the dictionary - what those are is what you define in the dict_slicer